processor/src/test/java/org/mapstruct/ap/test/fields/FieldsMappingTest.java

```diff
@@ -18,28 +18,36 @@
  */
 package org.mapstruct.ap.test.fields;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 import org.assertj.core.util.Lists;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mapstruct.ap.testutil.IssueKey;
 import org.mapstruct.ap.testutil.WithClasses;
 import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
 
-import static org.assertj.core.api.Assertions.assertThat;
-
 /**
  * @author Filip Hrisafov
  */
 @RunWith(AnnotationProcessorTestRunner.class)
 @IssueKey( "557" )
-@WithClasses( { Source.class, Target.class, SourceTargetMapper.class} )
+@WithClasses({ Source.class, Target.class, SourceTargetMapper.class, Invocation.class })
 public class FieldsMappingTest {
 
+    @Before
+    public void setUp() throws Exception {
+        Source.INVOCATIONS.clear();
```

Comment:
Instead of the invocations class, you could just modify the passed values in the setter/getter methods, eg. add 42 or mulitply with `-1` or so... 😉. Then you can avoid the copy of the `Invocation` class.